packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -594,45 +637,130 @@ describe('Table row selection', () => {
+                        initialSelection: ['0'],
+                        rowToClick: 0,
+                        clickModifiers: { shiftKey: true },
+                        expectedSelection: ['0'],
+                        fireEvent: false
+                    }
+                ];
+                const focused: string[] = [];
+                const disabled: string[] = [];
+                for (const configuration of configurations) {
+                    const specType = getSpecTypeByNamedList(
+                        configuration,
+                        focused,
+                        disabled
                     );
-                    expect(emittedIds).toEqual(expectedSelection);
-                });
-
-                it('clicking the already selected row maintains its selection and does not emit an event', async () => {
-                    await element.setSelectedRecordIds(['0']);
-                    await pageObject.clickRow(0);
+                    // eslint-disable-next-line @typescript-eslint/no-loop-func
+                    specType(configuration.name, async () => {
+                        await element.setSelectedRecordIds(
+                            configuration.initialSelection
+                        );
+                        await pageObject.clickRow(
+                            configuration.rowToClick,
+                            configuration.clickModifiers
+                        );
 
-                    const currentSelection = await element.getSelectedRecordIds();
-                    expect(currentSelection).toEqual(['0']);
-                    expect(selectionChangeListener.spy).not.toHaveBeenCalled();
-                });
+                        const currentSelection = await element.getSelectedRecordIds();
+                        expect(currentSelection).toEqual(
+                            jasmine.arrayWithExactContents(
+                                configuration.expectedSelection
+                            )
+                        );
+                        if (configuration.fireEvent) {
+                            expect(
+                                selectionChangeListener.spy
+                            ).toHaveBeenCalledTimes(1);
+                            const emittedIds = getEmittedRecordIdsFromSpy(
+                                selectionChangeListener.spy
+                            );
+                            expect(emittedIds).toEqual(
+                                jasmine.arrayWithExactContents(
+                                    configuration.expectedSelection
+                                )
+                            );
+                        } else {
+                            expect(
+                                selectionChangeListener.spy
+                            ).not.toHaveBeenCalled();
+                        }
```

Comment:
I'd really like to avoid conditional expects in the test code path. Some approaches to avoid that are to make a function that calculates results that can be expected on or to split the test up so a separate test can test the alternative code path.

I think splitting up the tests probably makes sense here.